submit-api/src/submit_api/schemas/activity_log.py

```diff
@@ -1,9 +1,66 @@
-"""Package ActivityLogSchema class.
+"""Activity Log Schema for Proponent and Staff Activity Logs."""
+from marshmallow import Schema, fields, post_dump
+from enum import Enum
 
-Manages the ActivityLogSchema
-"""
 
-from marshmallow import Schema, fields
+class ActionType(Enum):
+    """Enum for different activity actions."""
+
+    ORIGINAL_SUBMISSION = "Original Submission"
+    START_CONSULTATION_CHECK = "Start Consultation Check"
+    UPDATED_SUBMISSION_REQUESTED = "Updated Submission"
+    PASSED_CONSULTATION_CHECK = "Passed Consultation Check"
+    FAILED_CONSULTATION_CHECK = "Failed Consultation Check"
+    START_MP_REVIEW = "Start MP Review"
+    MP_ACCEPTED_APPROVED_SATISFIED = "MP Accepted/Approved/Satisfied"
+    MP_REVIEW_REJECTED = "MP Review Failed"
+    UPDATED_SUBMISSION = "Updated Submission"
+
+
+def get_activity_action(action: str, user_type: str) -> str:
+    """Map activity log actions based on user type (Proponent/Staff)."""
+    action_mapping = {
+        ActionType.ORIGINAL_SUBMISSION.value: {
+            "PROPONENT": "Original Submission",
+            "STAFF": "Original Submission"
+        },
+        ActionType.START_CONSULTATION_CHECK.value: {
+            "PROPONENT": "Start Consultation Check",
+            "STAFF": "Start Consultation Check"
+        },
+        ActionType.UPDATED_SUBMISSION_REQUESTED.value: {
+            "PROPONENT": "Update Requested",
+            "STAFF": "Update Requested"
+        },
+        ActionType.PASSED_CONSULTATION_CHECK.value: {
+            "PROPONENT": "Passed Consultation Check",
+            "STAFF": "Passed Consultation Check"
+        },
+        ActionType.FAILED_CONSULTATION_CHECK.value: {
+            "PROPONENT": "Revision Requested",
+            "STAFF": "Failed Consultation Check"
+        },
+        ActionType.START_MP_REVIEW.value: {
+            "PROPONENT": "Start MP Review",
+            "STAFF": "Start MP Review"
+        },
+        ActionType.MP_ACCEPTED_APPROVED_SATISFIED.value: {
+            "PROPONENT": "MP Accepted/Approved/Satisfied",
```

Comment:
these are individual. There one for accepted, one for approved and one for satisfied. We should not combine them